.github/workflows/main.yml

```diff
@@ -119,12 +119,12 @@ jobs:
       - uses: actions/checkout@v4
         with:
           submodules: recursive
-      - name: Set up Python ${{ matrix.python-version }}
+      - name: Set up Python
         uses: actions/setup-python@v5
         with:
-          python-version: ${{ matrix.python-version }}
+          python-version: "3.12"
```

Comment:
Why are we changing to only test 3.12?

Reply:
This is just the Python version that is used in the `gather` step for running _update_badges.py_ and NOT the one for running the tests. We could also keep the version unspecified, and the version of Python currently in `PATH` will be used, I guess.

Reply:
We want to get rid of this warning here: https://github.com/lightningd/plugins/actions/runs/9331506874
![image](https://github.com/lightningd/plugins/assets/157324775/b74406f9-032e-4af9-81e9-04acc15e583b)

Reply:
Ah ok, I see. So we want to keep `update_badges.py` in sync with the latest version of Python? What version is in the PATH? Is this not populating because `matrix.python_version` only exists in the `build-and-test` step and not in the `gather` step?

Reply:
Yes, `matrix.python_version` is outside the scope of the `gather` step and therefore the value for `python-version` was missing, as I see it. It is clearly a bug. Here is another log output to clarify:
![image](https://github.com/lightningd/plugins/assets/157324775/94d99712-dd86-4299-8281-025fc5273d41)

To fix this, I specified the latest stable Python version. 3.12. In fact, the recommended way is to set the Python version explicitly:
> The python-version input is optional. If not supplied, the action will try to resolve the version from the default .python-version file. If the .python-version file doesn't exist Python or PyPy version from the PATH will be used. The default version of Python or PyPy in PATH varies between runners and can be changed unexpectedly so we recommend always setting Python version explicitly using the python-version or python-version-file inputs.

https://github.com/actions/setup-python

Reply:
Ok sounds good!